src/util/tree-walker.test.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-env jest */
+
+import { getAllNodes, getRoot } from './tree-walker'
+
+describe('getAllNodes', () => {
+    test('should call the functions passed to it', async () => {
+        const getParent = jest.fn()
+        const getChildren = jest.fn().mockReturnValue([])
+        await getAllNodes({getParent, getChildren})({})
+        expect(getParent).toHaveBeenCalled()
+        expect(getChildren).toHaveBeenCalled()
+    })
+
+    test('should return all the nodes as an array', async () => {
+        const getParent = jest.fn().mockReturnValueOnce(
+            new Promise((resolve, reject) => {
```

Comment:
You can use `Promise.resolve({name: 'root'})` shortcut to make the code more concise.